### DISCUSSION TITLE
I cannot run the heat transfer module

### DISCUSSION THREAD

**Discussion by styyokuda** [idaholab/moose#29026] [Q&A Modules: General]
All,
I cannot run the heat transfer module. Pease see the attached error message and attached the Makefile in which the heat transfer module is set to be "yes." Please help.
S. Thomas
![Error message](https://github.com/user-attachments/assets/f3210881-de91-4afa-8bf8-e2cfd5639305)
![Makefile](https://github.com/user-attachments/assets/4054f238-c8d1-48bf-bb55-9a7147e7817c)

**Comment by GiudGiud** [comment]
Hello

when adding a new module you need to make clobberall then recompile. Sometimes you also need to clean moose as well

**Reply by styyokuda** [reply]
Thank you very much. 
I am using the Heat Transfer module included in MOOSE and I make clobberall before compile.
Please help.

**Reply by GiudGiud** [reply]
Go in moose/modules and make clobberall there

If it s still not working, save all your work in both folders using git commit, then use git clean -xfD to clean them

**Reply by styyokuda** [reply]
After "make clobberall" in moose/module, am I suppose to compile (make -j 10) in moose/module?
Please help.

**Reply by GiudGiud** [reply]
No go compile in your application.

**Reply by styyokuda** [reply]
"make clobberall" in the moose/module directory did not help ... sorry.

**Reply by GiudGiud** [reply]
time to switch to git clean -Xfd in moose/ and in your_app/ folders

**Reply by GiudGiud** [reply]
so
```
cd ~/project/moose
git add .
git commit -am "Save everything"
git clean -Xfd
cd ../my_app
git add .
git commit -am "Save everything"
git clean -Xfd
```
if you know what you want to save specifically, you can change the git add / git commit part

**Reply by styyokuda** [reply]
I did all of these git runs but it is still not working. It is seemingly caused by other things.

**Reply by GiudGiud** [reply]
ok you should git clone your app your app again and change the Makefile to see if it works that way

**Reply by styyokuda** [reply]
Thank you, GiudGiud!
So, I am going to do re-installation. I am going to move the current projects directory to a new name.
Also, do I have to delete any files before the re-installation?
Please help.

**Reply by GiudGiud** [reply]
I dont think so

**Reply by styyokuda** [reply]
All, 
I am having compilation errors in my Application directory (please see attached).
I do appreciate if anyone can help.
![Screenshot 2024-11-13 at 3 58 36 PM](https://github.com/user-attachments/assets/8e42cbe5-0973-420d-80a7-c43d480d27ba)

**Reply by GiudGiud** [reply]
why is there a "TRUE" macro? Did you add this to your app? Are you building your app with some other code?
